scripts/do-release.sh

```diff
@@ -272,6 +272,11 @@ function release_installers() {
     # Create a PR for subctl to use these versions
     local project="subctl"
     record_errors update_dependencies Subctl "${SUBCTL_CONSUMES[@]}"
+
+    # Create a PR in submariner-charts to update the CHARTS_VERSION in the Makefile
+    local project="submariner-charts"
+    local update_dependencies_extra=update_charts_versions
+    record_errors update_dependencies Charts
```

Comment:
Shouldn't this be done in the same place we create the operator update PR?

Otherwise you're updating the version already after the tag is created.

Also, this will only update it on the branch you're releasing on, not on `devel`, which is fine I guess unless the intention was to update this on `devel`

Reply:
> Shouldn't this be done in the same place we create the operator update PR?

I'm not sure where the "best" place is. I supposed it could be in the `projects` phase too but **submariner-charts** is an installer so the `installers` phase seems appropriate along with **subctl**.

> Otherwise you're updating the version already after the tag is created.

I'm not sure what you mean here. Are you referring to the **submariner-charts** tag? Isn't that created in the `released` phase?

> Also, this will only update it on the branch you're releasing on, not on `devel`, which is fine I guess unless the intention was to update this on `devel`

The intention is update the release branch.

Reply:
> > Shouldn't this be done in the same place we create the operator update PR?
> 
> I'm not sure where the "best" place is. I supposed it could be in the `projects` phase too but **submariner-charts** is an installer so the `installers` phase seems appropriate along with **subctl**.

Each stage is when those components get tagged, so on `installers` we tag `submariner-operator` and `submariner-charts` as those are out installers.
`subctl` is not an installer, it's a CLI, and as it relies on operator it has to be last, hence we create the relevant PR during the `installers` stage right after the operator is tagged and becomes available for `subctl` to use.

> 
> > Otherwise you're updating the version already after the tag is created.
> 
> I'm not sure what you mean here. Are you referring to the **submariner-charts** tag? Isn't that created in the `released` phase?

All these https://github.com/submariner-io/submariner-charts/tags are created on the `installers` stage, so since the chart "version" is actually the value of `CHARTS_VERSION`, we'd rather have it updated before the tag IIUC.

To put it differently - if I go and checkout tag `v0.15.1` in the project, which `CHARTS_VERSION` should I be seeing in the `Makefile`?

> 
> > Also, this will only update it on the branch you're releasing on, not on `devel`, which is fine I guess unless the intention was to update this on `devel`
> 
> The intention is update the release branch.

:+1:

Reply:
> subctl is not an installer, it's a CLI

It is a CLI installer, among other things. Otherwise, why did you name the phase "installers"? 😉 

> if I go and checkout tag v0.15.1 in the project, which CHARTS_VERSION should I be seeing in the Makefile?

_0.15.1_.  I now see that `INSTALLER_PROJECTS ` includes **submariner-charts** and in `release_installers`:

`for_every_project create_project_release "${INSTALLER_PROJECTS[@]}"`

I assumed the **submariner-charts** tag was created in the `released` phase. In `release_released`:

```
for_every_project create_project_release "${PROJECTS[@]}"
```

and `PROJECTS=(admiral cloud-prepare lighthouse shipyard subctl submariner submariner-charts submariner-operator)`.  So it creates a project release for **submariner-charts** et al in more than one phase but I see that `create_project_release` checks if it already exists.

So.... it seems to me we could either move creating of the **submariner-charts** to `projects` or don't create the **submariner-charts** in the projects phase, ie remove it from `INSTALLER_PROJECTS`. If all else being equal, I'd rather do the latter since IMO semantically the `installers` phase fits better.